comma/model.py

```diff
@@ -33,30 +38,103 @@ def setup(self, size: int, dir_params: str,
         self.dir_params = dir_params
         Hypothesis.validate_param_file(dir_params)
         if use_ipf:
-            self.agents = Individual.populate_ipf(size, dir_params)
+            self.agents = Individual.populate_ipf(size, dir_params, rng=self.rng)
         else:
-            self.agents = Individual.populate(size, dir_params)
+            self.agents = Individual.populate(size, dir_params, rng=self.rng)
+
+    def update_covid_counter(self):
+        """
+        Update the days_since_positive counter
+        for agents with covid_status of 1.
+        Returns:
+            None. This function updates the agent's covid counter in place.
+        """
+        for agent in self.agents:
+            if agent.covid_status == 1:
+                agent.days_since_positive += 1
+
+    def get_recovered_individuals(self) -> List[int]:
+        """
+        Get the indices of agents who are recovered from COVID-19
+
+        The function finds positive agents and checks if they are
+        recovered based on the number of days since they tested +ve.
+
+        Returns:
+            recovered (list): List of indices of recovered agents
+
+        """
+        positives = [
+            index for index, agent in enumerate(self.agents) if agent.covid_status == 1
+        ]
+        recovered = []
+        for i in positives:
+            days_positive = self.agents[i].days_since_positive
+
+            if self.agents[i].is_recovered(days_positive):
+                recovered.append(i)
```

Comment:
List comprehensions are a bit of a matter of taste, but here I see some potential for simplification:

```suggestion
        recovered = [
            i for i in positives
            if self.agents[i].is_recovered(self.agents[i].days_since_positive)
        ]
```

Reply:
Or, after our today's discussion (assuming a change in `Individual.is_recovered()`):

```suggestion                
        recovered = [
            agent
            for agent in self.agents
            if agent.is_recovered() and agent.covid_status
        ]
```

You could also consider changing `Individual.is_recovered()` so that it takes the `covid_status` into account too, simplifying the logic in this place.